### PR TITLE
feat: add babashka support

### PR DESCRIPTION
**Pull request by yqrashawn**
this PR is based on #3 

- add babashka support
- add optional `clj-type` parameter to register function
- run `clojure` or `bb` based on different `clj-type`

### CODE REVIEW

File: src/cloel.clj [hunk 1]
```diff
@@ -1,11 +1,10 @@
 (ns cloel
   (:require [clojure.edn :as edn])
-  (:import [java.net ServerSocket Socket]
-           [java.io BufferedReader InputStreamReader PrintWriter]
-           [java.util.concurrent ConcurrentHashMap]))
```
Comment (yqrashawn) [inline src/cloel.clj:5]: bb does not support `java.util.concurrent.ConcurrentHashMap`
I guess this has better performance over `atom`?

File: demo-bb/src/app.clj [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+(defn app-success [message]
+  ;; STEP 10: Send message to Emacs ASYNC.
+  (cloel/elisp-show-message message))
+
+(alter-var-root #'cloel/handle-client-message (constantly app-handle-client-message))
+(alter-var-root #'cloel/handle-client-connected (constantly app-handle-client-connected))
+
+(defn -main []
+  (cloel/start-server (Integer/parseInt (last *command-line-args*)))
+  (while true (Thread/sleep 1000)))
```
Comment (yqrashawn) [inline demo-bb/src/app.clj]: bb task will exist without this line

File: src/cloel.clj (hunk 1 above)
Comment (manateelazycat) [reply src/cloel.clj:5]: We chang `call-results` to atom ? 

call-results just hold function callback result, I think atom is enough.

File: demo-bb/src/app.clj (hunk 2 above)
Comment (manateelazycat) [reply demo-bb/src/app.clj]: Maybe promise is better way?

```
(defn -main []
  (cloel/start-server (Integer/parseInt (last *command-line-args*)))
  (.addShutdownHook (Runtime/getRuntime)
                    (Thread. #(println "Shutting down...")))
  @(promise))
```

**Pull request merged**